src/services/item/plugins/published/service.ts

```diff
@@ -114,6 +115,36 @@ export class ItemPublishedService {
     };
   }
 
+  private async getPublishedIfExist(item: Item, itemPublishedRepository: ItemPublishedRepository) {
+    try {
+      return await itemPublishedRepository.getForItem(item);
+    } catch (_e) {
+      return undefined;
+    }
+  }
+
+  async publishIfNotExist(
+    actor: Actor,
+    repositories: Repositories,
+    itemId: string,
+    { canBePrivate }: { canBePrivate?: boolean } = {},
+  ) {
+    if (!actor) {
+      throw new UnauthorizedMember(actor);
+    }
+    const { itemPublishedRepository } = repositories;
+
+    const item = await this.itemService.get(actor, repositories, itemId, PermissionLevel.Admin);
+
+    const itemPublished = await this.getPublishedIfExist(item, itemPublishedRepository);
+
+    if (itemPublished) {
+      return itemPublished;
+    }
+
+    return await this.post(actor, repositories, itemId, { canBePrivate });
```

Comment:
```suggestion
    return this.post(actor, repositories, itemId, { canBePrivate });
```

Reply:
Here too, I'm not sure to understand why I should not await @pyphilia